src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.ipp

```diff
@@ -927,9 +927,9 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
         uint32_t pincode;
 
         SuccessOrExit(error = ConfigurationMgr().GetSetupPinCode(pincode));
-        snprintf(pskd.m8, sizeof(pskd.m8) - 1, "%" PRIu32, pincode);
+        snprintf(pskd.m8, sizeof(pskd.m8) - 1, "%08" PRIu32, pincode);
```

Comment:
Why 8 digits?

2 ** 27 - 1 = 134217727

Reply:
According to previous discussions about QR code, the setup code seems to be an decimal number with 8 digits, and 99999999 needs 27 bits to store.

Reply:
The current QR-code draft spec is here: https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/qr_code/OnboardingPayload.adoc

Can you point to the relevant portion of the QR-code draft spec that indicates 8 decimal digits?

Reply:
Since the home kit's QR code is the baseline of this part, which also uses 8 digits and 27 bits field in qr code, I guess we will still using 8 digits here.
Another idea is just using 9 decimal digits since this is only relevalent to thread commissioning and made changes after the QR code spec update this part if necessary?

Reply:
@erjiaqing , the draft spec for [Encoding Method without Vendor and Product ID’s](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/qr_code/OnboardingPayload.adoc#encoding-method-without-vendor-and-product-ids) encodes all 27 bits of the PIN code without any constraint on range. Note that the manual pairing code includes more information than just the PIN code and is larger than 8 digits.